### PR TITLE
Use more pcl::Indices

### PR DESCRIPTION
**Pull request by mvieth**
Please see commit messages.

### CODE REVIEW

File: filters/src/passthrough.cpp [hunk 1]
```diff
@@ -297,8 +297,10 @@ pcl::PassThrough<pcl::PCLPointCloud2>::applyFilter (Indices &indices)
   indices.resize (indices_->size ());
   removed_indices_->resize (indices_->size ());
   int oii = 0, rii = 0;  // oii = output indices iterator, rii = removed indices iterator
-  const std::uint32_t xyz_offset[3] = {input_->fields[x_idx_].offset, input_->fields[y_idx_].offset, input_->fields[z_idx_].offset};
-  PCL_DEBUG ("[pcl::%s<pcl::PCLPointCloud2>::applyFilter] Field offsets: x: %u, y: %u, z: %u.\n", filter_name_.c_str (), xyz_offset[0], xyz_offset[1], xyz_offset[2]);
+  const auto x_offset = input_->fields[x_idx_].offset,
+             y_offset = input_->fields[y_idx_].offset,
+             z_offset = input_->fields[z_idx_].offset;
+  PCL_DEBUG ("[pcl::%s<pcl::PCLPointCloud2>::applyFilter] Field offsets: x: %u, y: %u, z: %u.\n", filter_name_.c_str (), x_offset, y_offset, z_offset);
```
Comment (kunaltyagi) [inline filters/src/passthrough.cpp:303]: Use %z here and cast {x,y,z}_offset to std::size

File: apps/src/openni_klt.cpp [hunk 2]
```diff
@@ -295,7 +295,7 @@ class OpenNIViewer {
   typename pcl::tracking::PyramidalKLTTracker<PointType>::Ptr tracker_;
   pcl::PointCloud<pcl::PointUV>::ConstPtr keypoints_;
   pcl::PointIndicesConstPtr points_;
-  pcl::PointIndicesConstPtr points_status_;
+  pcl::shared_ptr<const std::vector<int>> points_status_;
```
Comment (kunaltyagi) [inline apps/src/openni_klt.cpp:298]: This seems counter intuituve

File: registration/include/pcl/registration/impl/ia_ransac.hpp [hunk 3]
```diff
@@ -97,7 +97,7 @@ SampleConsensusInitialAlignment<PointSource, PointTarget, FeatureT>::selectSampl
   sample_indices.clear();
   while (static_cast<int>(sample_indices.size()) < nr_samples) {
     // Choose a sample at random
-    int sample_index = getRandomIndex(static_cast<int>(cloud.size()));
+    const auto sample_index = getRandomIndex(static_cast<int>(cloud.size()));
```
Comment (kunaltyagi) [inline registration/include/pcl/registration/impl/ia_ransac.hpp:100]: Do we need the static casts to `int`? Or the `int` type of `nr_samples`

**Review by kunaltyagi — COMMENTED** [review]
16/42 (It's a draft, but it's bound to be a large changeset, so I got cracking)

File: apps/src/openni_klt.cpp (hunk 2 above)
Comment (mvieth) [reply apps/src/openni_klt.cpp:298]: As described in the commit message, it has nothing to do with indices. The status can be 0, -1, or -2, describing whether the point was successfully tracked or not (and why not).

File: registration/src/correspondence_rejection_one_to_one.cpp [hunk 4]
```diff
@@ -53,7 +53,7 @@ pcl::registration::CorrespondenceRejectorOneToOne::getRemainingCorrespondences(
             pcl::registration::sortCorrespondencesByMatchIndexAndDistance());
 
   remaining_correspondences.resize(input.size());
-  int index_last = -1;
+  pcl::index_t index_last = -1;
```
Comment (kunaltyagi) [inline registration/src/correspondence_rejection_one_to_one.cpp:56]: static cast here since we're converting -1 (signed) to potentially unsigned

**Comment by kunaltyagi** [comment]
Thoughts on splitting the deprecation causing commit to another PR? It's fine here as well, but the PR topic might be confusing to see in the changelog

**Review by kunaltyagi — APPROVED** [review]
39/42. LGTM except the active conversations

File: registration/src/correspondence_rejection_one_to_one.cpp (hunk 4 above)
Comment (mvieth) [reply registration/src/correspondence_rejection_one_to_one.cpp]: Changed to `UNAVAILABLE` from pcl_base.h

File: registration/include/pcl/registration/impl/ia_ransac.hpp (hunk 3 above)
Comment (mvieth) [reply registration/include/pcl/registration/impl/ia_ransac.hpp]: I think I did what you asked. Please check if that is what you meant

**Comment by mvieth** [comment]
> Thoughts on splitting the deprecation causing commit to another PR? It's fine here as well, but the PR topic might be confusing to see in the changelog

Removed that commit. I will create another PR with those changes later.

**Review by larshg — APPROVED** [review]

**Comment by kunaltyagi** [comment]
🚀

**Pull request merged**